### PR TITLE
task #15 tested "GET /api/articles/:article_id/comments" endpoint

### PR DESCRIPTION
**Pull request by MariusVen**
task #15 tested "GET /api/articles/:article_id/comments" endpoint

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -225,4 +225,44 @@ describe("app", () => {
...
+                comment_id: expect.any(Number),
+                votes: expect.any(Number),
+                created_at: expect.any(String),
+                author: expect.any(String),
+                body: expect.any(String),
+              })
+            );
+          });
+        });
+    });
```
Comment (dav3rid) [inline __tests__/app.test.js:248]: Would also include a test for an empty array - article does exist but no associated comments

Comment (MariusVen) [reply __tests__/app.test.js:248]: Spend a lot of time to figure out.  My test is working now, but query is not a single line. So I am not sure if it's ok.

File: models/models.js [hunk 2]
```diff
@@ -67,7 +67,15 @@ exports.fetchArticles = () => {
...
-exports.fetchComments = (articleId) => {
+exports.fetchComments = async (articleId) => {
+  await db
+    .query("SELECT * FROM articles WHERE article_Id=$1", [articleId])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "No article found" });
+      }
+    });
+
```
Comment (maddierosecodes) [inline models/models.js]: Don't mix async/await with promises inside the same function & ideally not inside the same project!

File: models/models.js [hunk 3]
```diff
@@ -76,7 +84,7 @@ exports.fetchComments = (articleId) => {
     .then(({ rows }) => {
       const commentArray = rows;
       if (commentArray.length === 0) {
-        return Promise.reject({ status: 404, msg: "No article found" });
+        return [];
       }
       return commentArray;
     });
```
Comment (maddierosecodes) [inline models/models.js:81]: You're along the right lines with multiple DB queries being needed, but to accomplish this we would typically be returning a Promise.all inside the controller on two functions, one function being the model and the other being to check if the article id exists. Promise.all rejects when _any_ of the promises inside it rejects.

**Review by maddierosecodes — CHANGES_REQUESTED** [review]
Added some comments to explain how to refactor this :)
